### DISCUSSION TITLE
Pyroscope with Java

### DISCUSSION THREAD

**Discussion by maheshrajrp** [grafana/pyroscope#3154] [General]
Hi All,
I am trying Pyroscope with Java, I am able to successfully get information `process_cpu` from the application, however, I am targeting for more information on memory. Is it possible ? Does Pyroscope support it ?

Thanks

**Comment by korniltsev** [comment]
1. Assuming you are using  [java sdk](https://github.com/grafana/pyroscope-java/blob/main/README.md)
The documentation is here https://grafana.com/docs/pyroscope/latest/configure-client/language-sdks/java/

1.1 if you run the SDK as java agent , you can use `PYROSCOPE_PROFILER_ALLOC=512k` configuration to enable memory profiling
1.2 If you run the sdk as java library, you can use the `Config$Builder#setAlloc` method to enable allocations

2. If you use java profiling from grafana agent (`pyroscope.java` component), then you can use the `alloc` configuration option. (the docs are here https://grafana.com/docs/agent/latest/flow/reference/components/pyroscope.java/

```
pyroscope.java "java" {
  targets = discovery.relabel.java.output
  forward_to = [pyroscope.write.staging.receiver]
  profiling_config {
    interval = "60s"
    alloc = "512k"
    cpu = true
    sample_rate = 100
    lock = "1ms"
  }
}
```

**Comment by maheshrajrp** [comment]
Hi @korniltsev ,
Thanks for responding, I can see the following metrics now. Right, from Java perspective is there any other metrics we can fetch using Pyroscope Agent ? Have I missed anything ?

Is there any simple documentation that explains the possible metrics that can be scrapped  ? 

![image](https://github.com/grafana/pyroscope/assets/80001447/79be6cf0-2f6a-4ccb-8f4b-04e03d036075)

Thanks again

**Reply by Rperry2174** [reply]
> Hi @korniltsev ,
> 
> Thanks for responding, I can see the following metrics now. Right, from Java perspective is there any other metrics we can fetch using Pyroscope Agent ? Have I missed anything ?
> 
> 
> 
> Is there any simple documentation that explains the possible metrics that can be scrapped  ? 
> 
> 
> 
> ![image](https://github.com/grafana/pyroscope/assets/80001447/79be6cf0-2f6a-4ccb-8f4b-04e03d036075)
> 
> 
> 
> Thanks again
> 
> 

You should be able to see supported profile types in our docs [here](https://grafana.com/docs/grafana-cloud/monitor-applications/profiles/introduction/profiling-types/)

**Reply by maheshrajrp** [reply]
@Rperry2174  Thanks, this works!!
